websearch/brave/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+mod client;
+mod conversions;
+
+use std::cell::RefCell;
+
+use crate::client::{BraveSearchApi, SearchRequest};
+use crate::conversions::{params_to_request, response_to_results, validate_search_params};
+use golem_web_search::durability::Durablewebsearch;
+use golem_web_search::durability::ExtendedwebsearchGuest;
+use golem_web_search::golem::web_search::web_search::{
+    Guest, GuestSearchSession, SearchError, SearchMetadata, SearchParams, SearchResult,
+    SearchSession,
+};
+
+use golem_web_search::LOGGING_STATE;
+
+// Define a custom ReplayState struct
+#[derive(Debug, Clone, PartialEq, golem_rust::FromValueAndType, golem_rust::IntoValue)]
+pub struct BraveReplayState {
+    pub params: SearchParams,
```

Comment:
We need to track metadata in here (as otherwise metadata will always be None after the replay end).

minor: SearchParams do not need to be tracked in here, as we get them as part of start_search during replay. We could instead do
```
fn session_from_state(state: &Self::ReplayState, params: SearchParams) -> Result<Self::SearchSession, SearchError> ;
```